comment/models/comments.py

```diff
@@ -0,0 +1,59 @@
+from django.contrib.auth import get_user_model
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.db import models
+
+from comment.manager import CommentManager
+
+class Comment(models.Model):
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, default=None)
+    parent = models.ForeignKey('self', on_delete=models.CASCADE, null=True, blank=True)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+    content = models.TextField()
+    posted = models.DateTimeField(auto_now_add=True)
+    edited = models.DateTimeField(auto_now=True)
+
+    objects = CommentManager()
+
+    class Meta:
+        ordering = ['-posted', ]
+
+    def __str__(self):
+        if not self.parent:
+            return f'comment by {self.user}: {self.content[:20]}'
+        else:
+            return f'reply by {self.user}: {self.content[:20]}'
+
+    def __repr__(self):
+        return self.__str__()
+
+    def _get_reaction_count(self, reaction):
+        """
+        Get reaction count for a reaction
+
+        Args:
+            reaction (str): the reaction whose count is needed.
+
+        Returns:
+            int: count of the reaction
+        """
+        reaction_obj, _ = self.reactions.get_or_create(comment=self)
+        return getattr(reaction_obj, reaction, None)
+
+    @property
+    def replies(self):
+        return Comment.objects.filter(parent=self).order_by('posted')
+
+    @property
+    def is_edited(self):
+        return self.posted.timestamp() + 1 < self.edited.timestamp()
+
+    @property
+    def likes(self):
```

Comment:
In fact we don't need this property since we can get the reaction count straight forward:
```python
comment = Comment()
comment.reaction.likes
comment.raction.dislikes
```
Keeping them doesn't hurt.

Reply:
I have added these property so that these can be used directly in the templates without defining any other template tag or functions to send in the count. Just noticed the typo, will correct it too.